packages/test/src/test-integration-workflows.ts

```diff
@@ -222,3 +222,75 @@ test('Start of workflow with signal is delayed', async (t) => {
   const startDelay = workflowExecutionStartedEvent?.workflowExecutionStartedEventAttributes?.firstWorkflowTaskBackoff;
   t.is(tsToMs(startDelay), 4678000);
 });
+
+export async function executeEagerActivity(): Promise<string> {
+  // If eager activity dispatch is working, then the task will always be dispatched to the workflow
+  // worker. Otherwise, chances are 50%-50% for either workers. We execute the activity 20 times to
+  // make sure that the workflow worker is really getting the task thanks to eager activity
+  // dispatch, and not out of pure luck.
+  for (let i = 0; i < 20; i++) {
+    const res = await workflow
+      .proxyActivities({ scheduleToCloseTimeout: '5s', allowEagerDispatch: true })
+      .testActivity()
+      .catch(() => 'failed');
+    if (res === 'failed') return res;
+  }
+  return 'success';
+}
+
+test('Worker request Eager Activity Dispatch if possible', async (t) => {
```

Comment:
This doesn't test that the activity was eagerly dispatched.

I would run the two workers, start 3 activities (eager limit) and check the workflow worker handled the tasks.

Reply:
> This doesn't test that the activity was eagerly dispatched.

This test asserts that 20 activities executed one after the other all get dispatched to the workflow+activity worker, and none to the activity-worker. That is statistically very unlikely to happen unless this is due to eager activity dispatch. As a proof of that, if I disable Eager Execution

> I would run the two workers, start 3 activities (eager limit) and check the workflow worker handled the tasks.

Just to be sure, are you simply suggesting starting three activities at once (ie. `MAX_EAGER_ACTIVITY_RESERVATIONS_PER_WORKFLOW_TASK`), rather than one, or are you suggesting some other changes here?

Reply:
You're right, this is fine.
You could just schedule 3 at once, it's probably just as effective.